pkg/distributors/contracts/MerkleOrchard.sol

```diff
@@ -52,32 +52,28 @@ contract MerkleOrchard is IDistributor, Ownable {
         uint256 distribution;
         uint256 balance;
         address rewarder;
-        IERC20 rewardToken;
+        uint256 tokenIndex;
         bytes32[] merkleProof;
     }
 
     function _processClaims(
         address liquidityProvider,
         address recipient,
         Claim[] memory claims,
+        IERC20[] memory tokens,
         bool asInternalBalance
     ) internal {
-        // We want to keep track of the number of unique tokens so we can aggregate transfers
-        uint256 numRewardTokens;
-        IERC20[] memory rewardTokens = new IERC20[](claims.length);
-        uint256[] memory rewardAmounts = new uint256[](claims.length);
+        uint256[] memory amounts = new uint256[](tokens.length);
 
         Claim memory claim;
         for (uint256 i = 0; i < claims.length; i++) {
             claim = claims[i];
+            IERC20 token = tokens[claim.tokenIndex];
```

Comment:
```suggestion
        Claim memory claim;
        IERC20 token;
        for (uint256 i = 0; i < claims.length; i++) {
            claim = claims[i];
            token = tokens[claim.tokenIndex];
```

For consistency with `claim`. (Although sometimes I've found that doing this increases gas costs instead of decreasing them.)